model/EligibilityService.php

```diff
@@ -422,10 +426,21 @@ public function isManuallyAssigned()
     public function deliveryExecutionCreated(DeliveryExecutionCreated $event)
     {
         $monitoringService = $this->getServiceLocator()->get(DeliveryMonitoringService::SERVICE_ID);
-        $testCenter = $this->getTestCenter($event->getDeliveryExecution()->getDelivery(), $event->getUser());
+        /** @var DeliveryExecutionInterface $deliveryExecution */
+        $deliveryExecution = $event->getDeliveryExecution();
+        $testCenter = $this->getTestCenter($deliveryExecution->getDelivery(), $event->getUser());
         if (!empty($testCenter)) {
-            $deliverMonitoringData = $monitoringService->getData($event->getDeliveryExecution());
+            /** @var DeliveryMonitoringData $deliverMonitoringData */
+            $deliverMonitoringData = $monitoringService->getData($deliveryExecution);
             $deliverMonitoringData->update(TestCenterMonitoringService::TEST_CENTER_ID, $testCenter->getUri());
+            $executionContext = new DeliveryExecutionContext(
+                $deliveryExecution->getIdentifier(),
+                $testCenter->getUri(),
+                TcDeliveryExecutionContext::EXECUTION_CONTEXT_TYPE,
```

Comment:
What is the purpose to have different `DeliveryExecutionContext` implementations if all of them is the same and you need to inject the type of that?
I would say it would make more sense if you create a `new TcDeliveryExecutionContext()` and omit the `type` or just to have one implementation and keep the `setType()`. What do you think?

Reply:
I created TcDeliveryExecutionContext just to change constant value. Couldn't find more appropriate place for it. 
It does not make sense to create instance of TcDeliveryExecutionContext here, because it will be converted to json before storing in db. When we read it later we will crate general `DeliveryExecutionContext` object.